### ISSUE TITLE
bracktesting returns 0 values

### ISSUE THREAD

**Issue by Papyxulo** [freqtrade/freqtrade#7168]
## Describe your environment

  * Operating system: Ubuntu 22.04.1 LTS
  * Python Version: 3.10.4 (`python -V`)
  * CCXT version: 1.91.55 (`pip freeze | grep ccxt`)
  * Freqtrade Version: develop-9df10c6b5 (`freqtrade -V` or `docker-compose run --rm freqtrade -V` for Freqtrade running in docker)
  
## backtesting giving 0 values, What am i doing wrong ? 

I'm trying to run a strategy, but every time i try it it gives me 0 values, i've tested this behavior with multiple other strategies and all do the same. I assume i'm doing something wrong.

so i've gone back to basics 

1. Created a new strategy called `test1`  using `freqtrade new-strategy -s test1 --template minimal`
2. Downloaded data from binance using `freqtrade download-data -p .*/EUR -t 1d --timerange 20220101-20220801 --exchange binance`
3. Run strategy with `freqtrade backtesting --strategy test1 -i 1d --starting-balance 100 -c ./user_data/learndatasci-config.json -p .*/EUR --timerange 20220101-20220701`

Results : 

>Result for strategy test1
========================================================== BACKTESTING REPORT ==========================================================
|       Pair |   Buys |   Avg Profit % |   Cum Profit % |   Tot Profit EUR |   Tot Profit % |   Avg Duration |   Win  Draw  Loss  Win% |
|------------+--------+----------------+----------------+------------------+----------------+----------------+-------------------------|
|    BTC/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    ETH/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    XRP/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    SXP/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|   LINK/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    DOT/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    LTC/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    ADA/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    BCH/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    YFI/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    XLM/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    GRT/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    EOS/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|   DOGE/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|   EGLD/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|   AVAX/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    UNI/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    CHZ/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    ENJ/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|  MATIC/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|  THETA/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    HOT/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    WIN/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    VET/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    TRX/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|   SHIB/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    ETC/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    SOL/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    ICP/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|   RUNE/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|  LAZIO/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|  PORTO/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
| ALPINE/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|   ATOM/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|   GALA/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|   NEAR/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|  WAVES/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    APE/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    GMT/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    FTM/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    ZIL/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|  JASMY/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    GAL/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|    DAR/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|     OP/EUR |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
|      TOTAL |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
========================================================= ENTER TAG STATS =========================================================
|   TAG |   Buys |   Avg Profit % |   Cum Profit % |   Tot Profit EUR |   Tot Profit % |   Avg Duration |   Win  Draw  Loss  Win% |
|-------+--------+----------------+----------------+------------------+----------------+----------------+-------------------------|
| TOTAL |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
==================================================== EXIT REASON STATS =====================================================
|   Exit Reason |   Exits |   Win  Draws  Loss  Win% |   Avg Profit % |   Cum Profit % |   Tot Profit EUR |   Tot Profit % |
|---------------+---------+--------------------------+----------------+----------------+------------------+----------------|
===================================================== LEFT OPEN TRADES REPORT ======================================================
|   Pair |   Buys |   Avg Profit % |   Cum Profit % |   Tot Profit EUR |   Tot Profit % |   Avg Duration |   Win  Draw  Loss  Win% |
|--------+--------+----------------+----------------+------------------+----------------+----------------+-------------------------|
|  TOTAL |      0 |           0.00 |           0.00 |                0 |           0.00 |           0:00 |     0     0     0     0 |
============================== SUMMARY METRICS ==============================
No trades made. Your starting balance was 100 EUR, and your stake was 10 EUR.
=============================================================================

Strategy: 

```
# pragma pylint: disable=missing-docstring, invalid-name, pointless-string-statement
# flake8: noqa: F401

# --- Do not remove these libs ---
import numpy as np  # noqa
import pandas as pd  # noqa
from pandas import DataFrame  # noqa
from datetime import datetime  # noqa
from typing import Optional, Union  # noqa

from freqtrade.strategy import (BooleanParameter, CategoricalParameter, DecimalParameter,
                                IStrategy, IntParameter)

# --------------------------------
# Add your lib to import here
import talib.abstract as ta
import pandas_ta as pta
import freqtrade.vendor.qtpylib.indicators as qtpylib


class test1(IStrategy):
    """
    This is a strategy template to get you started.
    More information in https://www.freqtrade.io/en/latest/strategy-customization/

    You can:
        :return: a Dataframe with all mandatory indicators for the strategies
    - Rename the class name (Do not forget to update class_name)
    - Add any methods you want to build your strategy
    - Add any lib you need to build your strategy

    You must keep:
    - the lib in the section "Do not remove these libs"
    - the methods: populate_indicators, populate_entry_trend, populate_exit_trend
    You should keep:
    - timeframe, minimal_roi, stoploss, trailing_*
    """
    # Strategy interface version - allow new iterations of the strategy interface.
    # Check the documentation or the Sample strategy to get the latest version.
    INTERFACE_VERSION = 3

    # Optimal timeframe for the strategy.
    timeframe = '5m'

    # Can this strategy go short?
    can_short: bool = False

    # Minimal ROI designed for the strategy.
    # This attribute will be overridden if the config file contains "minimal_roi".
    minimal_roi = {
        "60": 0.01,
        "30": 0.02,
        "0": 0.04
    }

    # Optimal stoploss designed for the strategy.
    # This attribute will be overridden if the config file contains "stoploss".
    stoploss = -0.10

    # Trailing stoploss
    trailing_stop = False
    # trailing_only_offset_is_reached = False
    # trailing_stop_positive = 0.01
    # trailing_stop_positive_offset = 0.0  # Disabled / not configured

    # Run "populate_indicators()" only for new candle.
    process_only_new_candles = True

    # These values can be overridden in the config.
    use_exit_signal = True
    exit_profit_only = False
    ignore_roi_if_entry_signal = False

    # Number of candles the strategy requires before producing valid signals
    startup_candle_count: int = 30

    # Strategy parameters
    buy_rsi = IntParameter(10, 40, default=30, space="buy")
    sell_rsi = IntParameter(60, 90, default=70, space="sell")

    # Optional order type mapping.
    order_types = {
        'entry': 'limit',
        'exit': 'limit',
        'stoploss': 'market',
        'stoploss_on_exchange': False
    }

    # Optional order time in force.
    order_time_in_force = {
        'entry': 'gtc',
        'exit': 'gtc'
    }
    

    def informative_pairs(self):
        """
        Define additional, informative pair/interval combinations to be cached from the exchange.
        These pair/interval combinations are non-tradeable, unless they are part
        of the whitelist as well.
        For more information, please consult the documentation
        :return: List of tuples in the format (pair, interval)
            Sample: return [("ETH/USDT", "5m"),
                            ("BTC/USDT", "15m"),
                            ]
        """
        return []

    def populate_indicators(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        """
        Adds several different TA indicators to the given DataFrame

        Performance Note: For the best performance be frugal on the number of indicators
        you are using. Let uncomment only the indicator you are using in your strategies
        or your hyperopt configuration, otherwise you will waste your memory and CPU usage.
        :param dataframe: Dataframe with data from the exchange
        :param metadata: Additional information, like the currently traded pair
        :return: a Dataframe with all mandatory indicators for the strategies
        """
        
        # Momentum Indicators
        # ------------------------------------

        # RSI
        dataframe['rsi'] = ta.RSI(dataframe)

        # Retrieve best bid and best ask from the orderbook
        # ------------------------------------
        """
        # first check if dataprovider is available
        if self.dp:
            if self.dp.runmode.value in ('live', 'dry_run'):
                ob = self.dp.orderbook(metadata['pair'], 1)
                dataframe['best_bid'] = ob['bids'][0][0]
                dataframe['best_ask'] = ob['asks'][0][0]
        """

        return dataframe

    def populate_entry_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        """
        Based on TA indicators, populates the entry signal for the given dataframe
        :param dataframe: DataFrame
        :param metadata: Additional information, like the currently traded pair
        :return: DataFrame with entry columns populated
        """
        dataframe.loc[
            (
                (qtpylib.crossed_above(dataframe['rsi'], self.buy_rsi.value)) &  # Signal: RSI crosses above buy_rsi
                (dataframe['volume'] > 0)  # Make sure Volume is not 0
            ),
            'enter_long'] = 1
        # Uncomment to use shorts (Only used in futures/margin mode. Check the documentation for more info)
        """
        dataframe.loc[
            (
                (qtpylib.crossed_above(dataframe['rsi'], self.sell_rsi.value)) &  # Signal: RSI crosses above sell_rsi
                (dataframe['volume'] > 0)  # Make sure Volume is not 0
            ),
            'enter_short'] = 1
        """

        return dataframe

    def populate_exit_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        """
        Based on TA indicators, populates the exit signal for the given dataframe
        :param dataframe: DataFrame
        :param metadata: Additional information, like the currently traded pair
        :return: DataFrame with exit columns populated
        """
        dataframe.loc[
            (
                (qtpylib.crossed_above(dataframe['rsi'], self.sell_rsi.value)) &  # Signal: RSI crosses above sell_rsi
                (dataframe['volume'] > 0)  # Make sure Volume is not 0
            ),
            'exit_long'] = 1
        # Uncomment to use shorts (Only used in futures/margin mode. Check the documentation for more info)
        """
        dataframe.loc[
            (
                (qtpylib.crossed_above(dataframe['rsi'], self.buy_rsi.value)) &  # Signal: RSI crosses above buy_rsi
                (dataframe['volume'] > 0)  # Make sure Volume is not 0
            ),
            'exit_short'] = 1
        """
        return dataframe

```

**Comment by stash86** [comment]
Most of the time, it's because you are setting wrong stake amount

**Comment by xmatthias** [comment]
Assuming binance EUR limits are similar to the USDT limits, then 10€ is not sufficient, as it'll be below the lower limit of what's tradable on binance.

For USDT, the limit is ~15USDT for most pairs - but i'd recommend to go to 20 USDT - to leave a bit more room for pairs with recent moves.

**Comment by Papyxulo** [comment]
> Assuming binance EUR limits are similar to the USDT limits, then 10€ is not sufficient, as it'll be below the lower limit of what's tradable on binance.
> 
> For USDT, the limit is ~15USDT for most pairs - but i'd recommend to go to 20 USDT - to leave a bit more room for pairs with recent moves.

That did the trick, thank you
